staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -96,7 +96,7 @@ func (recorder *recorderImpl) makeEvent(refRegarding *v1.ObjectReference, refRel
 	}
 	return &eventsv1.Event{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%v.%x", refRegarding.Name, t.UnixNano()),
+			Name:      util.GenerateEventName(refRegarding.Name, t.UnixNano()),
```

Comment:
I think we wanted to pass the event key here: https://github.com/kubernetes/kubernetes/blob/d99d3f7eb734dc9284a90374fcc6d8cc32d1946d/staging/src/k8s.io/client-go/tools/events/event_broadcaster.go#L58-L66 instead of refRegarding.Name + t.UnixNano() as we can be certain that the key is unique to each event

Reply:
I tried that and I discarded because I don't like the idea of generating the event object first and after generate the name,  
```
event := Event{} \\ without name
name := genKey(event)
event.Name = name
```

with this approach we have the same behavior as today plus we support the objects names that are not supported by the event name validation, it is more easier to think about, if we want unique key then just we can use a sha or uuid directly, no need to get the key

Reply:
@dgrisonnet the event key is not guaranteed a valid event.name